packages/@ember/-internals/runtime/lib/mixins/action_handler.d.ts

```diff
@@ -1,3 +1,6 @@
 import { Mixin } from '@ember/-internals/metal';
 
-export default class ActionHandler extends Mixin {}
+export default class ActionHandler extends Mixin {
+  actions?: Record<string, (...args: unknown[]) => void>;
```

Comment:
1. Does it *have* to return `void` (i.e. will anything else be thrown away anyway)? I think the answer is no, it can return anything. The type-checker will more or less shrug at it I think, but it should probably be `unknown`.

2. I don't think the arguments will type check in practice either. 🤔 [playground][p] For types like this, `unknown[]` is usually wrong for arguments, because it means subclasses can never have *narrower* types in the function parameters than than, because it breaks assignability/Liskov substitution/variance rules (whichever way is easiest to think about it for you; they all boil down to the same issue here).

Net, the type should *probably* be:

```suggestion
  actions?: Record<string, (...args: any[]) => unknown>;
```

(Technically, `...args: never[]` is the "right" answer here, because all functions are valid super types of that, since `never` is the "bottom" type… but that confuses the heck out of people. And reasonably so.)

[p]: https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEkoGdnwIJgC4EsD2AdgBJQHAQgzwDeAUPPFNvgcgPwBc8ASuHjMAA8yLDBwEA5gBp4ACgB0i2BORcArgQDWBPAHcCAbQC6ASngBeAHzwAbnhzBLAbloBfWrSSp4AMTx54EAAPLBAyNExcQhIyCio6BiYo1gsaegZ4CTxZAigAWxAuETFJGSgJQvgCNTyAI0ozBIyGOCw1GAJ4AAMAEmpcgtd4HDQ+8pAhgE8QWDQ8CGAul2b3Bnd3IA